### ISSUE TITLE
FNX-529 ⁃ [Dark Theme] Follow-up UX edits

### ISSUE THREAD

**Issue by AmyYLee** [mozilla-mobile/fenix#2271]
Hi, 
The original [dark theme bug](https://github.com/mozilla-mobile/fenix/issues/826) was closed before these edits were completed. Opening a new bug to track these edits as part of the implementation review. 

## Acceptance Criteria
* [ ] Update colors for the filter buttons ("Shortcuts") and "x" in the text bar, matching the mocks.
* [ ] Add a screenshot the the PR and mention Amy

![Screen Shot 2019-10-02 at 11 43 11 AM](https://user-images.githubusercontent.com/5992746/66059331-dc008a00-e509-11e9-8074-3ce7788dc047.png)

Dark Theme Edits
https://mozilla.invisionapp.com/share/JYRD2QPUK8R

**Comment by AmyYLee** [comment]
@vesta0 I think this should be marked as P1 since these affects accessibility if colours fail contrast tests.

**Comment by mheubusch** [comment]
Nope - it's a P2.  Accessibility is a priority but as long as the default light theme supports accessible contrast then this bug can't come before other  more important work.

**Comment by vesta0** [comment]
@AmyYLee can we further clarify the requirements in this bug for engineering?

**Comment by AmyYLee** [comment]
@vesta0 I reviewed the invision doc and I've removed outdated material. Please see comments in the invision for the outstanding edits. Thanks!

**Comment by vesta0** [comment]
Thanks @AmyYLee !
@liuche are the requested edits clear for engineering?

**Comment by liuche** [comment]
@ekager since you did Dark Theme, could you take a quick look and make sure these look reasonable? Then @vesta0 can prioritize it in her Feature Backlog next time for sprint planning.

**Comment by ekager** [comment]
I checked the invision doc and these are the only two remaining actionable things I can see (but @AmyYLee please correct me if I'm wrong).
Anyone can pick these up:

@AmyYLee I'm not sure exactly what this one is referring to and I'm not sure anyone else has access to the spec could you clarify?
<img width="603" alt="Screen Shot 2019-08-01 at 8 02 40 PM" src="https://user-images.githubusercontent.com/25442310/62341234-5a539800-b497-11e9-9a6c-f244370b5ddc.png">
<img width="434" alt="Screen Shot 2019-08-01 at 7 59 52 PM" src="https://user-images.githubusercontent.com/25442310/62341235-5aec2e80-b497-11e9-8ff1-d10ccafb00f2.png">

**Comment by AmyYLee** [comment]
> I checked the invision doc and these are the only two remaining actionable things I can see (but @AmyYLee please correct me if I'm wrong).
> Anyone can pick these up:
> 
> @AmyYLee I'm not sure exactly what this one is referring to and I'm not sure anyone else has access to the spec could you clarify?
> <img alt="Screen Shot 2019-08-01 at 8 02 40 PM" width="603" src="https://user-images.githubusercontent.com/25442310/62341234-5a539800-b497-11e9-9a6c-f244370b5ddc.png">
> <img alt="Screen Shot 2019-08-01 at 7 59 52 PM" width="434" src="https://user-images.githubusercontent.com/25442310/62341235-5aec2e80-b497-11e9-8ff1-d10ccafb00f2.png">

Hi, 

@ekager Here are the colour edits for the filter states. Let me know if you have any questions.

<img width="608" alt="Screen Shot 2019-08-09 at 2 32 47 PM" src="https://user-images.githubusercontent.com/5992746/62800992-be3b1980-bab2-11e9-99c6-adbfe5403fea.png">

**Comment by vesta0** [comment]
Whoever picks up this ticket, talk to @AmyYLee about the colour scheme!

**Comment by liuche** [comment]
Let's wait until 9/30 when @AmyYLee comes back, to clarify what Filter* states are.

**Comment by AmyYLee** [comment]
@ekager @liuche Hi, these are what I mean by filter states. 
![Screen Shot 2019-10-02 at 11 43 11 AM](https://user-images.githubusercontent.com/5992746/66059331-dc008a00-e509-11e9-8074-3ce7788dc047.png)

**Comment by liuche** [comment]
Sounds like this is ready to be worked on - this just needs to get updated colors for the filter buttons and "x" in the text bar, matching the mocks.

**Comment by lime124** [comment]
removing @AmyYLee from assigned to signal it's free to be picked up.

**Comment by liuche** [comment]
Quick question @AmyYLee: these dividers seem really hard to see on Dark Mode. Is there a different color that we could use? (Also, these icons push the url further off the screen - do they need to be that big because of touch targets, or could they be smaller?)

<img width="325" alt="Screen Shot 2019-10-29 at 10 24 32 AM" src="https://user-images.githubusercontent.com/1082253/67792585-a1214180-fa36-11e9-9d4a-084b5457a901.png">
<img width="345" alt="Screen Shot 2019-10-29 at 10 25 56 AM" src="https://user-images.githubusercontent.com/1082253/67792590-a2eb0500-fa36-11e9-813a-396357bd95d0.png">

**Comment by AmyYLee** [comment]
Hi,
I think the reason it's hard to see is that the shadow is missing in the dark theme version or the shadow is set for light theme so it doesn't show up in dark. The icon touch targets are set to be 40dp wide so wee need a minimum distance between icons. Let me know if you have any more questions.

It should be the following for Dark theme

Hairline #FBFBFE 12% opacity
Shadow #FBFBFE 12% opacity, Offset 1dp 

![Dark](https://user-images.githubusercontent.com/5992746/67879867-0eef6b00-fb14-11e9-99b0-ede778738971.png)

**Comment by liuche** [comment]
QA needed for dark mode:
- [ ] correct divider opacity
- [ ] Shortcuts button color
- [ ] "x" button color in toolbar

**Comment by abodea** [comment]
Hello, I retested this issue on the latest Nightly build from 12/11, for more details please check the following:
#### As I understand from all the comments above:

| Mode | Divider opacity | Shortcuts button color| "x" button color in toolbar | Hairline, and Shadow|
| ------------- |:-------------:|:-------------:| :-------------:| :-------------:|
| Normal | ✅   | ✅|  ✅|  ✅
| Private |  ✅  | ✅ | ✅|  ❌



>Normal Browsing with Dark Theme ON
![Screenshot_20191211-114651_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611159-67cf1a80-1c0d-11ea-9e93-378cab0a9a7f.jpg)
![Screenshot_20191211-114757_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611300-a49b1180-1c0d-11ea-838d-de2dd0665f71.jpg)
![Screenshot_20191211-114806_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611355-b8df0e80-1c0d-11ea-8904-24147dbe44cb.jpg)
![Screenshot_20191211-114702_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611423-dca25480-1c0d-11ea-892b-bb88e69f109a.jpg)

>Private Browsing 
![Screenshot_20191211-114725_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611775-6baf6c80-1c0e-11ea-9bdd-34a56df6558f.jpg)
![Screenshot_20191211-114818_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611783-6d793000-1c0e-11ea-8c81-8c40e189e5cd.jpg)
![Screenshot_20191210-101810_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70611843-8386f080-1c0e-11ea-958f-09f94f61a35e.jpg)
![Screenshot_20191211-114721_Firefox Nightly](https://user-images.githubusercontent.com/42831109/70612919-75d26a80-1c10-11ea-8dd3-9029e4f15e4e.jpg)

In the last picture from PB mode, it can be seen that the hairline is exactly how it was in Normal browsing before the changes.
Besides the hairline from PB mode, everything looks good.

@liuche @AmyYLee should the hairline for PB be fixed like in Normal browsing and if possible to confirm that my verification is valid, I looked in the code for the opacity but compared with an older version I can see the improvement.
After it will be decided regarding the PB hairline, I will close this bug.
I will remove the qa:needed until further notice.

**Comment by liuche** [comment]
@AmyYLee should there be a difference between PB mode in dark vs light theme? specifically for the hairline urlbar divider? It doesn't seem to be covered in the mocks (which handle normal browsing mode only).

**Comment by AmyYLee** [comment]
> @AmyYLee should there be a difference between PB mode in dark vs light theme? specifically for the hairline urlbar divider? It doesn't seem to be covered in the mocks (which handle normal browsing mode only).

Hi, 

@abodea @liuche It should be #FBFBFE @16% opacity. PBM is the same for Dark/Light themes. Thanks

**Comment by abodea** [comment]
Verified as fixed on the latest **Nightly 3/5** for more information please check the following:
#### Devices:
• **Google Pixel 3a XL (Android 10)**
• **Samsung Galaxy S10+ (Android 9)**
• **Huawei Mate 20 Lite (Android 8.1.0)**
• **Samsung Galaxy S7 (Android 7.0.0)**
• **OnePlus A3 (Android 6.0.1)**
• **LG Nexus 4 (Android 5.1.1)**

##### Note that I verified this issue for both cases **top** and **bottom** URL bar for `Normal, Private` mode with `dark/light` themes and the hairline is correctly displayed.
